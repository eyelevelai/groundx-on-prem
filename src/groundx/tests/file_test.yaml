suite: "file renders as expected"
values:
  - ../values.yaml

tests:
  - it: minio-operator.yaml
    templates:
      - ../charts/file/templates/operator-deployment.yaml
    asserts:
      - matchSnapshot: {}
  - it: file_tenant.yaml
    templates:
      - ../templates/services/file_tenant.yaml
    asserts:
      - matchSnapshot: {}
