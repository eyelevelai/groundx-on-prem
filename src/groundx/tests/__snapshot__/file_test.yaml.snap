file_tenant.yaml:
  1: |
    apiVersion: minio.min.io/v2
    kind: Tenant
    metadata:
      annotations:
        helm.sh/hook: post-install,post-upgrade
        helm.sh/hook-delete-policy: before-hook-creation
        helm.sh/hook-weight: "10"
      labels:
        app: file-tenant
        appVersion: 0.1.0
        chart: groundx-0.1.0
        heritage: Helm
        version: 0.1.0
      name: file-tenant
      namespace: eyelevel
    spec:
      tenant:
        certificate:
          requestAutoCert: false
        configSecret:
          accessKey: minio
          secretKey: minio123
        name: file-tenant
        nodeSelector:
          node: eyelevel-cpu-only
        resources:
          requests:
            cpu: 100m
            memory: 256Mi
        securityContext:
          allowPrivilegeEscalation: false
          capabilities:
            drop:
              - ALL
          fsGroup: 1000
          fsGroupChangePolicy: OnRootMismatch
          runAsGroup: 1000
          runAsNonRoot: true
          runAsUser: 1000
          seccompProfile:
            type: RuntimeDefault
        servers: 1
        size: 20Gi
        volumesPerServer: 1
minio-operator.yaml:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/version: v6.0.3
        helm.sh/chart: file-6.0.3
      name: minio-operator
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: file
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: file
            app.kubernetes.io/version: v6.0.3
            helm.sh/chart: file-6.0.3
        spec:
          affinity:
            podAntiAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                - labelSelector:
                    matchExpressions:
                      - key: name
                        operator: In
                        values:
                          - minio-operator
                  topologyKey: kubernetes.io/hostname
          containers:
            - args:
                - controller
              env:
                - name: OPERATOR_STS_ENABLED
                  value: "on"
              image: quay.io/minio/operator:v6.0.3
              imagePullPolicy: IfNotPresent
              name: file
              resources:
                requests:
                  cpu: 200m
                  ephemeral-storage: 500Mi
                  memory: 256Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                runAsGroup: 1000
                runAsNonRoot: true
                runAsUser: 1000
                seccompProfile:
                  type: RuntimeDefault
          nodeSelector:
            node: eyelevel-cpu-only
          securityContext:
            fsGroup: 1000
            runAsGroup: 1000
            runAsNonRoot: true
            runAsUser: 1000
          serviceAccountName: minio-operator
